dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py

```diff
@@ -156,7 +157,8 @@ def __init__(
         self.resource_list = resource_list or []
         self.dependence = dependence or {}
         self.wait_start_timeout = wait_start_timeout or {}
-        self.condition_result = condition_result or self.DEFAULT_CONDITION_RESULT
+        if task_type != TaskType.CONDITIONS:
+            self.condition_result = condition_result or self.DEFAULT_CONDITION_RESULT
```

Comment:
Maybe you can also define a function decorated with `@condition_result.setter ` in `Conditions` task.